cmd/observe/flows_filter.go

```diff
@@ -163,6 +241,60 @@ func (of *flowFilter) checkConflict(t *filterTracker) error {
 	return nil
 }
 
+// checkInconsistentNamespaces checks if the namespaces in pods and services make sense
+// i.e. it checks that we don't request a service in one namespace with pods in another namespace
+func checkInconsistentNamespaces(pods, services []string) error {
+	for _, pod := range pods {
+		podNs := namespaceFromName(pod)
+		if podNs == "" {
+			continue
+		}
+		for _, svc := range services {
+			if ns := namespaceFromName(svc); podNs != ns {
+				return fmt.Errorf("namespace of service %q conflict with pod %q", svc, podNs)
+			}
+		}
+	}
+	return nil
+}
+
+// checkNamespaceConflicts checks for conflicts in namespaces, pods and services
+func checkNamespaceConflicts(t *filterTracker) error {
+	var errs []error
+
+	if t.left != nil {
+		errs = append(errs,
+			t.ns.conflicts(t.left.SourcePod),
```

Comment:
trying to wrap my head around this check, specifically why we don't have `t.ns.conflicts(t.left.DestinationPod)` here.

Since `--{from,to}-pod` is incompatible with `--namespace` (i.e. the flag setting `t.ns`), then when `t.left.SourcePod` is non empty it was set through `--pod` and then `t.left.SourcePod == t.right.DestinationPod`, and both `t.left.DestinationPod` and `t.right.SourcePod` are empty right? If correct, I would be tempted to still check `t.ns.conflicts(t.left.DestinationPod)` (and respectively for svc and right) to make this codepath less reliant on understanding the different flag conflict / filter filling logic, since `conflicts()` will always return `nil` when given an empty slice, but I don't feel strongly about it.

Reply:
Yes. My reasoning was that both `left.DestinationPod` and `right.SourcePod` should be empty. But you're right there is no good reason not to check it anyways.